service/controller/key/key.go

```diff
@@ -308,6 +310,26 @@ func DNSZones(customObject providerv1alpha1.AzureConfig) providerv1alpha1.AzureC
 	return customObject.Spec.Azure.DNSZones
 }
 
+func InstanceIDFromNode(node v1.Node) (string, error) {
+	// Extract "000001" from "nodepool-w8kjg-000001".
+	var base36 string
+	{
+		m := regexp.MustCompile(`-[^-]{6}$`)
```

Comment:
This could be better when moved to package-global variable:
- The regex needs to be compiled only once because it's static.
- `MustCompile()` is good, but it should be executed when binary is started to get the panic immediately instead of only triggering when there's unhealthy node.